### ISSUE TITLE
Allow plugin assets with the `.mjs` extension

### ISSUE THREAD

**Issue by danburzo** [k-next/kirby#5463]
## Description

It would be helpful if plugin assets with the `.mjs` extension would be symlinked to the `media/` folder. This has been brought up before here:

- https://github.com/getkirby/kirby/issues/2792
- https://kirby.nolt.io/294

And on first glance this PR sounded like it would have fixed it by including `mjs` among the matched file extensions:

- https://github.com/getkirby/kirby/pull/3444

However, I am not able to make it work.

**Expected behavior**  

With a plugin containing the assets `assets/index.js`, `assets/index.mjs`, and `assets/index.css`, these should all work in `site/templates/default.php`:

```html
<script src='/media/plugins/provider/plugin-name/test.js'></script>
<script type='module' src='/media/plugins/provider/plugin-name/test.mjs'></script>
<link rel='stylesheet' src='/media/plugins/provider/plugin-name/test.css'>
```

However, `test.mjs` is not loaded.

## To reproduce

Here’s the plugin, for convenience: [plugin-name.zip](https://github.com/getkirby/kirby/files/12269269/plugin-name.zip)

When altering the pattern in `routes.php` to remove `js` (and leaving just `mjs`), the `test.mjs` file is symlinked and loaded correctly: 

```php
<?php
[
  'pattern' => $media . '/plugins/(:any)/(:any)/(:all).(css|map|gif|mjs|jpg|png|svg|webp|avif|woff2|woff|json)'
]
```

(Not sure what that piece of info means exactly, but maybe it helps?)

## Your setup
**Kirby Version**  

Kirby 3.9.6.1, local development using PHP 8.2.8’s built-in web server.

**Comment by afbora** [comment]
I've found the issue and solution. But I need to make sure that solution is correct. You can test from here: https://regex101.com/r/5nL69X/1

### Solution 1

I don't why but the issue is unescaped dot (`.`).  Should be `\.`. Currently `mjs` detected as `js` extension.

With solution:
https://regex101.com/r/2vW9UR/1

````diff
- 'pattern' => $media . '/plugins/(:any)/(:any)/(:all).(css|map|gif|js|mjs|jpg|png|svg|webp|avif|woff2|woff|json)',
+ 'pattern' => $media . '/plugins/(:any)/(:any)/(:all)\.(css|map|gif|js|mjs|jpg|png|svg|webp|avif|woff2|woff|json)',
````
https://github.com/getkirby/kirby/blob/3.9.6.1/config/routes.php#L63

### Solution 2

Again I don't know why (🙈) but using ungreedy as `U` identifier fixes the issue.

````diff
- if (preg_match('#^' . $this->regex($pattern) . '$#u', $path, $parameters)) {
+ if (preg_match('#^' . $this->regex($pattern) . '$#Uu', $path, $parameters)) {
````
https://github.com/getkirby/kirby/blob/3.9.6.1/src/Http/Route.php#L186

Could you confirm the solution, please? @lukasbestle @distantnative

**Comment by afbora** [comment]
Is this issue related with [Plugin::assets: support associative array](https://github.com/getkirby/kirby/commit/c54adffdc01424c60e06ecf1a1d7120681f0cc72) PR/commit? May be this issue can be fixed with that commit?

**Comment by lukasbestle** [comment]
@afbora Escaping the dot is the correct solution. I think it would be good as a first fix for 3.9.7. The commit you linked is for #5247 and therefore a bit of a larger change.

To explain why escaping the dot works (as well as using ungreedy capture, but not as clean): The filename capture group with `(.*)` is greedy and captures as many characters as it can while still making the overall match work. So it also captures the dot in the `test.mjs` filename. The dot in the regex then matches any character (here: the `m`). And finally the last capture group captures `js`. Making the capture group ungreedy reverses this behavior and happens to solve the issue. But escaping the dot fully solves the issue as it clearly separates the two capture groups with the fixed separator we are looking for.

**Comment by danburzo** [comment]
Oh wow, I totally missed that the dot was unescaped all this time. Good catch, @afbora!
